core/server/models/plugins/has-posts.js

```diff
@@ -0,0 +1,61 @@
+const _ = require('lodash');
+const _debug = require('ghost-ignition').debug._base;
+const debug = _debug('ghost-query');
+
+const addHasPostsWhere = function () {
+    const tableName = _.result(this, 'tableName');
+    const comparisonField = `${tableName}.id`;
+
+    this.query(function (qb) {
+        return qb.whereIn(comparisonField, function () {
+            const innerQb = this
+                .distinct('posts_authors.author_id')
+                .select()
+                .from('posts_authors')
+                .whereRaw(`posts_authors.author_id = ${comparisonField}`)
+                .join('posts', 'posts.id', 'posts_authors.post_id')
+                .andWhere('posts.page', '=', false)
```

Comment:
IMO authors can write posts or pages.

Reply:
Hm... I was basing thing on the filter applied [here](https://github.com/TryGhost/Ghost/blob/master/core/server/models/plugins/include-count.js#L22-L23). Not sure which way it should be done. But It makes sense that any kind of publication should count. Should we include `unpublished` as well?

Reply:
no only published 👍